### DISCUSSION TITLE
Convert Integer to byteslice, and then use as keys for maps on Algorand

### DISCUSSION THREAD

**Discussion by asolpshinning** [reach-sh/reach-lang#1211] [Q&A] [answered]
@jeapostrophe - You may not be notified on Discord.
But a question was asked in Algorand's Discord about when UInt key for maps would be supported. As you can see below, someone from Algorand answered there is no plan for that but that integer can be converted to byte slice. I am not sure how all this works but will reach be able to support conversion of UInt to byte slice and then allow use of byte slice as keys for maps ?
Thanks!!!

![image](https://user-images.githubusercontent.com/57355937/176234331-32a8ff99-f34e-4c69-b2d1-902a297df1d2.png)

**Comment by jeapostrophe** [comment, marked as answer]
I have no idea who `barnji` is and what he is talking about. When Algorand adds "box storage", Reach will support all datatypes as keys, from `UInt` to `Tuple(UInt, Contract)` and beyond

**Reply by barnjamin** [reply]
I am ~~spartacus~~ barnji

The only question I answered was whether or not it was technically possible to use an int as a key. It is not possible but it is of course possible to convert it to bytes.

@asolpshinning my above answer has no bearing on whether or not automatic conversion of an int to bytes should be a supported feature of reach

**Reply by jchaselubitz** [reply]
@jeapostrophe @barnjamin, thank you both for taking the time to discuss this. It looks like I could have better-asked my original question in Discord. Is there any indication of when Algorand will add box storage? We are trying to determine whether to wait for that or to approach the whole problem differently.

**Reply by jeapostrophe** [reply]
At AVM in Austin two weeks ago, they said it will happen this summer.

**Reply by kcelestinomaria** [reply]
Hello @jeapostrophe , just hoping onto this thread as I asked a similar question just today. Since the box storage feature for Maps is not available on Algo yet, what are the best data structure alternatives to use to counter this? Tried with my team to think around that limitation and no luck until now(at least the 2nd week on the problem). I remember you advising Maps are the best for use cases like NFT data, are there anything else someone can use that has the best traits or close to Maps? Thanks.

**Reply by kcelestinomaria** [reply]
There is also an issue where when you add new values to a Map in consensus, it overwrites the previous entry you added if it came from the same address

**Reply by jeapostrophe** [reply]
I answered on the other thread. Specifically on the question of NFT attributes, suppose I have the NFT "Jay McCarthy" and I want to extend it with "height" and "charm". One design is to have the "Jay McCarthy" contract have a map of these attributes and an external agent set them. Alternatively, I could launch a new contract called "Jay McCarthy's charm" and then a client will search the set of all contracts for ones that mention "Jay McCarthy" when they get displayed. That will be tedious and more expensive, but it will work, if you can't wait for more Algorand features.

**Reply by kcelestinomaria** [reply]
I think I'd rather wait @jeapostrophe for the update, thanks for the explanation, let me try comprehending to depth the Map's explanation you gave. Thanks again.

**Comment by jchaselubitz** [comment]
Algorand just released box storage!! https://github.com/algorand/go-algorand/releases/tag/v3.12.0-beta
